org-noter/org-noter
# Improve org-roam integration.

## Problem

Our org-roam integration allows for any file to be used to store notes. The user is prompted when `M-x org-noter` is invoked from a PDFView.

The downside of this approach is that the user has to remember/search where the notes were taken last time.

Apart from un-necessary cognitive load this could lead to multiple notes files.

## Solution

Using org-roam, check if there's a node that contains `NOTER_DOCUMENT` with a specified pdf file. If there is one, open it. Otherwise user standard "org-roam-node-find" completing read.

```mermaid
flowchart TD
    A[User invokes M-x org-noter on a PDF file] --> B{Has user previously taken notes for the specific PDF file?}
    B -- Yes --> C[Open the notes file automatically]
    B -- No --> D[Ask the user to select a notes file interactively]
    D --> E[Open the selected notes file]
```

## Checklist


## Steps to Test

1. org-roam installed and org-roam integration enabled (`org-noter-enable-org-roam-integration`).
2. Open a PDF document for which you have no notes.
3. `M-x org-noter`
4. Specify an org-roam node to store the notes in.
5. Complete org-noter session (`org-noter-kill-session`)
6. Reopen your PDF document and invoke org-noter again.

You should be dropped into your notes without being asked to select a node.
